Created Test Command for Shooter

StephenDay1 opened the pull request:
At this point I don't think it will get the information for these, but they appear on shuffleboard. I don't know how to get the actual information from the motors

StephenDay1 committed:
Created Test Command for Shooter

At this point I don't think it will get the information for these, but they appear on shuffleboard. I don't know how to get the actual information from the motors

StephenDay1 commented:
This is for #43.

devinvs commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -49,6 +49,7 @@ public RobotContainer() {
 
     // shooter.setDefaultCommand(new ShooterTest(shooter));
     // intake.setDefaultCommand(new IntakeTest(intake));
+    shooter.setDefaultCommand(new ShooterTestCommand(shooter));
```
We don't want this to be the default command...

StephenDay1 replied:
Ok, I just did that trying to test the command, it was having problems and I was trying to make sure the command was being called. I meant to get rid of it when I was done.

devinvs commented on src/main/java/frc/robot/commands/ShooterTestCommand.java:
```diff
@@ -0,0 +1,118 @@
+    if ((int)input_bottom_motor.getNumber(0) > 5500) {
+      input_bottom_motor.setNumber(5500);
+    } else {
+      input_bottom_motor.getNumber(0);
```
You need to be doing a lot more here. First off, you need to get all the values from the shuffleboard and _use them to control the subsystem_. Second, you need to be posting data here, which for neo's will be a combination of methods from the CANEncoder and the CANSparkMax classes.

devinvs commented on src/main/java/frc/robot/commands/ShooterTestCommand.java:
```diff
@@ -0,0 +1,118 @@
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+  }
```
Can you close the shuffleboard tab here? Also, set all motor values back to zero here, essentially resetting the robot.

StephenDay1 committed:
Worked on ShooterTestCommand #43

I am not done yet still. I just want you to look at this because everything was appearing on shuffleboard, and all of the sudden, now it's not.  If you can figure out why that would help.

I changed MotorPlacement to public, I don't know if I was supposed to do that, but I used it in the TestCommand.

devinvs commented on src/main/java/frc/robot/commands/ShooterTestCommand.java:
```diff
@@ -95,19 +100,36 @@ public void initialize() {
+    bottom_motor_current.setNumber(0);
+    input_top_motor.setNumber(0);
+    input_bottom_motor.setNumber(0);
   }
```
I don't want you to set all the shuffleboard fields to zero, I want you to stop all the motors.

devinvs commented on src/main/java/frc/robot/commands/ShooterTestCommand.java:
```diff
@@ -0,0 +1,140 @@
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    top_motor_velocity.setNumber(0);
+    bottom_motor_velocity.setNumber(0);
+    top_motor_position.setNumber(0);
+    bottom_motor_velocity.setNumber(0);
+    top_motor_current.setNumber(0);
+    bottom_motor_current.setNumber(0);
+    input_top_motor.setNumber(0);
+    input_bottom_motor.setNumber(0);
```
Don't set the shuffleboard fields to zero, actually stop the motors.

StephenDay1 committed:
Made it so motors actually turn off in Shooter

Sorry, I didn't realize you made this request until today.

StephenDay1 committed:
Merge branch 'develop' into feature/test-commands
